### ISSUE TITLE
Just having a look.

### ISSUE THREAD

**Issue by MovingtoMars** [0xbadb002/goof#1]
Interesting project so far. How are you going to go about register allocation? Is this intended as an intermediate language for compilers?

**Comment by 0xbadb002** [comment]
Cheers. As for register allocation, I'm not too sure. I was trying a few algorithms, one of which was called "backtracking", which is really slow. There's another algorithm I was going to try, though I forgot it's name. For now I'm implementing the basics and doing the register allocation last.

It's not really an IR, more like a random language for learning new things. The initial idea was to make a syntax that was really simple and can be easily translated into SSA. Though I kind of got carried away and wanted to try make some weird C-like language.

**Comment by MovingtoMars** [comment]
One way you could explore is to create a representation of assembly code. Assign the operands to virtual registers, then assign those virtual registers to hardware registers using interference analysis (and spill some virtual regs to stack if necessary).

**Comment by 0xbadb002** [comment]
@MovingtoMars Sounds interesting

**Comment by MovingtoMars** [comment]
Hey, just letting you know I'm working on a project you may be interested in. It's sort of a codegen library in Rust for stuff like register allocation, instruction selection, assembly analysis and so on. I'm wondering if you're interested at all.

**Comment by felixangell** [comment]
@MovingtoMars  Sounds cool, got a link I'll check it out?

**Comment by MovingtoMars** [comment]
I haven't uploaded it yet. It's still very early stages and I'm still figuring out how a lot of it's going to work. I can upload it how it is soon if you like.

**Comment by felixangell** [comment]
No rush, I can't guarantee I'll hop onto it right away :wink: But whenever you feel like it or have something, give it an upload and drop me a link and I'll check it out. :smile:

**Comment by MovingtoMars** [comment]
Do you come on IRC lately?

**Comment by felixangell** [comment]
Not too much, but you can add me on Discord if you have one: Felix#1553

**Comment by MovingtoMars** [comment]
I sent you a friend request

**Comment by felixangell** [comment]
Cool, I'll add you up but I'm off to sleep for tonight. Going to delete the other comments just for privacies and things... :tada:

**Comment by MovingtoMars** [comment]
Did you get my request?

**Comment by felixangell** [comment]
I don't seem to have anything in my pending box, are you sure you got the number right? #1553

**Comment by MovingtoMars** [comment]
hmm, just tried again

**Comment by felixangell** [comment]
Nope... what's your name#id? I'm going to sleep soon but I'll add you quickly before then

**Comment by MovingtoMars** [comment]
mvtmrs#3425

**Comment by felixangell** [comment]
@MovingtoMars Sent :)
